src/test/java/com/iota/iri/hash/KerlTest.java

```diff
@@ -70,26 +70,36 @@ public void loopRandTritsFromBigInt() throws Exception {
         int[] inTrits;
         byte[] bytes = new byte[Kerl.BYTE_HASH_LENGTH];
         int[] outTrits = new int[Kerl.HASH_LENGTH];
-        for (int i = 0; i<10_000; i++) {
+        for (int i = 0; i < 10_000; i++) {
             inTrits = getRandomTrits(trit_size);
             inTrits[242] = 0;
 
             BigInteger in_bigInteger = Kerl.bigIntFromTrits(inTrits, 0, trit_size);
-            Kerl.bytesFromBigInt(in_bigInteger,bytes,0);
-            BigInteger out_bigInteger = Kerl.bigIntFromBytes(bytes,0,bytes.length);
+            Kerl.bytesFromBigInt(in_bigInteger, bytes);
+            BigInteger out_bigInteger = Kerl.bigIntFromBytes(bytes, 0, bytes.length);
             Kerl.tritsFromBigInt(out_bigInteger, outTrits, 0, trit_size);
 
-            if(i % 1_000 == 0) {
-                System.out.println(String.format("%d iteration: %s",i, in_bigInteger ));
+            if (i % 1_000 == 0) {
+                System.out.println(String.format("%d iteration: %s", i, in_bigInteger));
             }
-            Assert.assertTrue(String.format("bigInt that failed: %s",in_bigInteger),Arrays.equals(inTrits,outTrits));
+            Assert.assertTrue(String.format("bigInt that failed: %s", in_bigInteger), Arrays.equals(inTrits, outTrits));
         }
     }
 
+    @Test
+    public void limitBigIntFromTrits() {
+        // this confirms that the limit we set using long math works for any size input
+        int[] trits = new int[2048];
+        Arrays.fill(trits, 1);
+        Kerl.bigIntFromTrits(trits, 0, trits.length);
+        Arrays.fill(trits, -1);
+        Kerl.bigIntFromTrits(trits, 0, trits.length);
+    }
+
```

Comment:
This is the one added test. Nothing fancy - just proves that with a big input, the overflow does not happen.

Reply:
Hmmm, looking again at this test, I am uncertain if it is good for anything.
BigInteger doesn't produce overflow exceptions as far as I know.

The only way this test can fail is with `OutOfMemoryException`.
Am I missing anything?
If so do tell, else remove it please.

Reply:
I will have to revise the test to be more explicit. Basically, this has a long math routine in it that needs to be checked for overflow:

```
BigInteger value = BigInteger.ZERO;
        for (int n = offset + size - 1; n >= offset; ) {
            int count = 0;
            long num = 0L;
            while (n >= offset && count < MAX_POWERS_LONG) {
                num = 3 * num + trits[n--];
                count++;
            }
            value = value.multiply(RADIX_POWERS[count]).add(BigInteger.valueOf(num));
        }
        return value;
```

The correct test would prove that trit[MAX_POWERS_LONG] produces no overflow in `num = 3 * num + trits[n--]`

The current test proves that in a not-so-clear fashion. As the boundary between MAX_POWERS_LONG is crossed, the  BigInteger result should always be the same regardless of the type of math used.

The trit[2048] test proves that ... but it just needs to be changed to trit[MAX_POWERS_LONG];

Reply:
With the assertion this test makes more sense since `int`s don't throw overflow exceptions ;-)